### ISSUE TITLE
How do Scanners work

### ISSUE THREAD

**Issue by antonTan96** [nus-cs2103-AY2324S1/forum#9]
```
        Scanner sc = new Scanner(System.in);
        System.out.println(sc.next());
        sc.close();
        sc = new Scanner(System.in);
        System.out.println(sc.next());
```

Apparently you can only close Scanners once.
when I tried running the code above, i would be returned with a NoSuchElementException.
Why doesn't declaring a new Scanner reopen the Scanner?

**Comment by damithc** [comment]
Good question @antonTan96 Let's see if anyone in the class can shed some light on this.

**Comment by HugeNoob** [comment]
> Why doesn't declaring a new Scanner reopen the Scanner?

Read: https://stackoverflow.com/questions/58244954/why-cant-i-just-create-another-scanner-object-after-using-scanner-close

TLDR: It seems that sc.close() closes the `System.in` input stream, which means you can't recreate a new Scanner from it again.

**Comment by antonTan96** [comment]
Interesting. There also seems to be no way to reopen the `System.in` input stream. Thanks for the info!
